addons/purchase/tests/test_purchase_invoice.py

```diff
@@ -102,6 +102,40 @@ def test_vendor_bill_delivered(self):
             self.assertEqual(line.qty_to_invoice, 0.0)
             self.assertEqual(line.qty_invoiced, 5)
 
+    def test_vendor_bill_analytic_account_default_change(self):
+        """Tests whether, when an analytic account rule is set, and user changes manually the analytic account on the po,
+            it is the same that is mentioned in the bill."""
+        AnalyticAccount = self.env['account.analytic.account']
+        analytic_account_default = AnalyticAccount.create({
+            'name': 'default',
+        })
+        analytic_account_manual = AnalyticAccount.create({
+            'name': 'manual',
+        })
+        analytic_rule = self.env['account.analytic.account'].create({
+            'analytic_id': analytic_account_default.id,
+            'product_id': self.product_order.id,
+        })
+
+        purchase_order = self.env['purchase.order'].with_context(tracking_disable=True).create({
+            'partner_id': self.partner_a.id,
+        })
+        PurchaseOrderLine = self.env['purchase.order.line'].with_context(tracking_disable=True)
+        pol_prod_order = PurchaseOrderLine.create({
```

Comment:
If it's a manual change, you should use the Form class to simulate what a user could do and avoid an onchange break in the test for the futur